lib/_http_server.js

```diff
@@ -331,17 +289,30 @@ function connectionListener(socket) {
     parser.maxHeaderPairs = 2000;
   }
 
-  socket.addListener('error', socketOnError);
-  socket.addListener('close', serverSocketCloseListener);
-  parser.onIncoming = parserOnIncoming;
-  socket.on('end', socketOnEnd);
-  socket.on('data', socketOnData);
+  var state = {
+    onData: null,
+    onError: null,
+    onEnd: null,
+    onClose: null,
+    outgoing: [],
+    incoming: [],
+    outgoingData: 0
+  };
+  state.onData = socketOnData.bind(undefined, this, socket, parser, state);
+  state.onError = socketOnError.bind(undefined, this, socket, state);
+  state.onEnd = socketOnEnd.bind(undefined, this, socket, parser, state);
+  state.onClose = socketOnClose.bind(undefined, socket, state);
```

Comment:
If you were referring to the use of `undefined` instead of `socket`, that's because I actually noticed a perf difference when the `this` parameter was set to anything (including `null`), so I opted to bind function parameters instead.